### PR TITLE
Update README TOC

### PR DESCRIPTION
**Pull request by JCZuurmond**
Forgot to do this in #1607

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,11 +35,29 @@ See [contributing instructions](CONTRIBUTING.md) to help improve this project.
...
+        * [Step 2.2: Create/Modify Cloud Principals and Credentials](#step-22-createmodify-cloud-principals-and-credentials)
+        * [Step 2.3: Create External Locations](#step-23-create-external-locations)
+        * [Step 2.4: Create "Uber Principal"](#step-24-create-uber-principal)
+        * [Step 2.5: Create Catalogs and Schemas](#step-25-create-catalogs-and-schemas)
+    * [Step 3: Upgrade the Metastore](#step-3-upgrade-the-metastore)
+    * [Step 4: Odds and Ends](#step-4-odds-and-ends)
+      * [Step 4.1: Skipping Table/Schema](#step-41-skipping-tableschema)
+      * [Step 4.2: Moving objects](#step-42-moving-objects)
+      * [Step 4.2: Aliasing objects](#step-42-aliasing-objects)
+      * [Step 4.3: Reverting objects](#step-43-reverting-objects)
```
Comment (asnare) [inline README.md:55]: These aren't indented consistently with the earlier steps and need to be a level deeper.

Comment (asnare) [inline README.md:55]: I'm not sure these belong in the ToC which is supposed to be an overview. 

Although there are #anchors available for every heading, that doesn't mean the ToC needs to and arguably these headings are content (and not headings for the content).

File: README.md [hunk 2]
```diff
@@ -35,11 +35,29 @@ See [contributing instructions](CONTRIBUTING.md) to help improve this project.
...
     * [Post Migration Data Reconciliation Task](#post-migration-data-reconciliation-task)
     * [Other considerations](#other-considerations)
+    * [[EXPERIMENTAL] Scan tables in mounts Workflow](#experimental-scan-tables-in-mounts-workflow)
+      * [<b>Always run this workflow AFTER the assessment has finished</b>](#balways-run-this-workflow-after-the-assessment-has-finishedb)
```
Comment (asnare) [inline README.md:59]: This is like the steps above; I'm not sure it should be in the ToC.

**Review by asnare — COMMENTED** [review]
Some suggestions to consider, depending on whether they're accepted or not one thing that needs to be fixed.

File: README.md (hunk 1 above)
Comment (JCZuurmond) [reply README.md:55]: The TOC is autogenerated includes all subheadings. The autogeneration has my preference over manually excluding the `.x` steps, though, if could configure it we can exclude this level

Comment (JCZuurmond) [reply README.md:55]: What do you mean? They are a lever deeper right?

Comment (asnare) [reply README.md:55]: They're at the wrong indent-level: step 3 and 4 should be at the same level as 1 and 2, but they're not.

I didn't realise this was autogenerated, sorry. It means that the heading-level is wrong for steps 3 and 4 (as well as 4.x).

**Review by asnare — APPROVED** [review]

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**